### PR TITLE
[SPARK-49120][BUILD] Bump Gson 2.11.0

### PR DESCRIPTION
**Pull request by pan3793**
### What changes were proposed in this pull request?

Currently, Spark pulls Gson 2.2.4 from `hive-exec`, which is pretty old and [vulnerable](https://cve.mitre.org/cgi-bin/cvename.cgi?name=CVE-2022-25647), this PR proposes to upgrade it to the latest version 2.11.0.

image

### Why are the changes needed?

For security.

### Does this PR introduce _any_ user-facing change?

No.

### How was this patch tested?

GHA.

### Was this patch authored or co-authored using generative AI tooling?

No.

### CODE REVIEW

**Comment by pan3793** [comment]
cc @dongjoon-hyun @yaooqinn @LuciferYang

File: dev/deps/spark-deps-hadoop-3-hive-2.3 [hunk 1]
```diff
@@ -62,11 +62,12 @@ derby/10.16.1.1//derby-10.16.1.1.jar
 derbyshared/10.16.1.1//derbyshared-10.16.1.1.jar
 derbytools/10.16.1.1//derbytools-10.16.1.1.jar
 dropwizard-metrics-hadoop-metrics2-reporter/0.1.2//dropwizard-metrics-hadoop-metrics2-reporter-0.1.2.jar
+error_prone_annotations/2.18.0//error_prone_annotations-2.18.0.jar
```
Comment (yaooqinn) [inline dev/deps/spark-deps-hadoop-3-hive-2.3:65]: please include this in the LICENSE-binary

Comment (LuciferYang) [inline dev/deps/spark-deps-hadoop-3-hive-2.3:65]: Do we have to add this dependency?

Comment (pan3793) [reply dev/deps/spark-deps-hadoop-3-hive-2.3:65]: short answer, no, but Guava changed it to compile scope in https://github.com/google/gson/pull/2346

I'm okay with either include or exclude it from Spark classpath

Comment (yaooqinn) [reply dev/deps/spark-deps-hadoop-3-hive-2.3:65]: +1 for removal

**Review by yaooqinn — APPROVED** [review]

**Review by LuciferYang — APPROVED** [review]

**Pull request closed without merging**